content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,285 @@
+
+### Chai
+
+JavaScript kata should use [Chai](https://www.chaijs.com/) as the assertion library.
+
+:::note
+For simplicity, through the rest of this article it's assumed that the `chai` package is imported and `assert` is extracted from it with the following statement:
+
+```javascript
+const chai = require('chai');
+const assert = chai.assert;
+```
+
+This is a convention used in many JavaScript kata, but it's not a requirement, and authors can choose to import the package in any way they find suitable for them. Also, the `should` or `expect` style can also be used from `chai`.
+:::
+
+The most commonly used assertions include:
+
+- [`assert.strictEqual(actual, expected[, message])`](https://www.chaijs.com/api/assert/#method_strictequal) - Asserts that `actual === expected`.
+- [`assert.deepEqual(actual, expected[, message])`](https://www.chaijs.com/api/assert/#method_deepequal) - Asserts that two arrays or objects have the same values. `strictEqual` does not work since arrays and objects are reference types in JS.
+- [`assert.closeTo(actual, expected, delta[, message])`](https://www.chaijs.com/api/assert/#method_closeto) - Asserts that `abs(actual - expected) <= delta`. Should be used when comparing floating-point numbers.
+
+See the [Chai docs](https://www.chaijs.com/api/assert/) for all the methods.
+
+:::note
+The [`truncateThreshold` option](https://www.chaijs.com/guide/styles/#configtruncatethreshold) controls how arrays are stringified when `expected` and `actual` values are prepared for display in assertion messages. Setting it to 0 provides the most detailed feedback even for large arrays, but then assertion messages can flood the console and make the output difficult to read.
+
+```javascript
+chai.config.truncateThreshold = 0;
+```
+
+:::
+
+### Test feedback
+
+Unlike Python or Ruby's test framework, Mocha handles combines multiple assertions in an `it` block into just one. For example, if all the assertions pass, then it will just show one `Test Passed`, and not a `Test Passed` for every single assertion. And if a test fails, it will not show all the other tests before/after it that passed/failed, it will only show the first failed test.
+
+### Generating random numbers
+
+JavaScript provides a simple [`Math.random`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random) function that returns a random floating-point number between `0` and `1`. You can use this to generate random numbers by multiplying it by another number, like `5`, and then flooring it, which returns a random number between 0 and **4**. Here's a simple utility to generate a random number between a lower and upper limit, inclusive:
```

Comment:
> which returns a random number between 0 and **4**.

Is unstyled "0" intentional?

![image](https://user-images.githubusercontent.com/639336/103630624-c9d47f00-4ef6-11eb-8aad-63f4a04769be.png)

The strong **4** makes sense. 0 should be `0` like others for consistency?